e2e-tests/README.md

```diff
@@ -0,0 +1,189 @@
+
+If you have multiple versions of Firefox installed, you can specifiy a specific one by using the `--firefox-path <path to firefox binary>` flag.
+
+___Running tests against localhost___
+
+	$ py.test --driver Firefox --base-url http://localhost:8000 tests/
+
+___Running the tests on stage___
+
+	$ py.test --driver Firefox tests/
+
+___Running tests against production___
+
+	$ py.test --driver Firefox --base-url https://crash-stats.mozilla.com tests/
+
+___Running specific tests___
+
+You can run tests in a given file::
+
+    $ py.test --driver Firefox tests/desktop/test_search.py
+
+You can run tests that match a specific name:
+
+    $ py.test --driver Firefox -k test_search_for_unrealistic_data
+
+You can run tests whose names match a specific pattern:
+
+    $ py.test --driver Firefox -k test_search
+
+__Output__
+
+Output of a test run should look something like this:
+
+    ============================= test session starts ==============================
+    platform linux2 -- Python 2.7.3 -- pytest-2.2.4
+    collected 73 items
+
+    tests/test_crash_reports.py .........................xx..........x.xx....x.
+    tests/test_layout.py xx
+    tests/test_search.py .........x.x.
+    tests/test_smoke_tests.py ...........
+
+    ============== 63 passed, 10 xfailed in 970.27 seconds ===============
+
+__Note__
+"~" will not resolve to the home directory when used in the py.test command line.
+
+The pytest plugin that we use for running tests has a number of advanced
+command line options available. To see the options available, run
+`py.test --help`. The full documentation for the plugin can be found
+[here][pytest-selenium].
+
+__Troubleshooting__
+
+If the test run hangs with Firefox open but no URL gets entered in the address
+box, some combinations of the Firefox version, and the python Selenium bindings
+version may not be compatible. Upgrading each of them to latest often fixes it.
+
+Tips and tricks
+---------------
+
+Because Selenium opens real browser windows, it will steal focus and switch
+workspaces. Firefox doesn't have a headless mode of operation, so we can't
+simply turn off the UI.
+
+__Use a different driver__
+
+[pytest-selenium] provides the ability to run tests against [many other][test envs] browser environments -- consider using a different driver executable or external provider.
+
+__xvfb__
+
+Xvfb provides a fairly easily work around on Linux.
+
+
+On Linux:
+
+    Install Xvfb and run the tests with it's xvfb-run binary. For
+    example, if you run tests like::
+
+        $ py.test ...
+
+
+    You can switch to this to run with Xvfb::
+
+        $ xvfb-run py.test ...
+
+
+    This creates a virtual X session for Firefox to run in, and sets
+    up all the fiddly environment variables to get this working
+    well. The tests will run as normal, and no windows will open, if
+    all is working right.
+
+
+Getting involved as a contributor
+---------------------------------
+
```

Comment:
This whole section is nicely written and awesome but it just doesn't make sense now that this README isn't about the project but a sub-section of the code.